lib/compat.php

```diff
@@ -132,8 +132,8 @@ function gutenberg_ensure_wp_api_request() {
  */
 function gutenberg_disable_editor_settings_wpautop( $settings ) {
 	$post = get_post();
-	if ( is_object( $post ) && gutenberg_post_has_blocks( $post->ID ) ) {
```

Comment:
Is this intentionally reverting changes from #2817?

Reply:
Not intentional. The only intended change there was to pass the `$post` object to `gutenberg_post_has_blocks()` instead of `$post->ID`. 

(`get_post()` accepts either the post ID or the post object, if we already have the object it's better/faster to pass it instead of just the ID).